### ISSUE TITLE
Dartdoc failing with `UnimplementedError: No members on Library.sentinel are accessible`

### ISSUE THREAD

**Issue by sigurdm** [dart-lang/dartdoc#3826]
On this package https://pub.dev/packages/walletconnect_modal_flutter

```
OUT: Documenting walletconnect_modal_flutter... Discovering libraries... Linking elements... ERR: dartdoc 8.0.10 (/tmp/analyzerBIFJCG/pub-cache/global_packages/dartdoc/bin/dartdoc.dart-3.4.3.snapshot) failed: UnimplementedError: No members on Library.sentinel are accessible #0 _LibrarySentinel.noSuchMethod (package:dartdoc/src/model/library.dart:18:7) dart-lang/pub-dev#1 _LibrarySentinel.element (package:dartdoc/src/model/library.dart:15:7) dart-lang/pub-dev#2 SpecialClass.matchesClass (package:dartdoc/src/special_elements.dart:55:28) dart-lang/pub-dev#3 SpecialClasses.addSpecial.
```

![image](https://github.com/user-attachments/assets/fee119f1-2557-4d84-ae62-cabf423b5b3c)

We should probably also improve the message when dartdoc is failing. It is kind'a hard to follow

**Comment by isoos** [comment]
More logs from https://pub.dev/packages/walletconnect_modal_flutter/score/log.txt:

```
2024-08-02 02:02:02.810582 WARNING: dartdoc 8.0.10 (/tmp/analyzerAIZJTO/pub-cache/global_packages/dartdoc/bin/dartdoc.dart-3.4.3.snapshot) failed: UnimplementedError: No members on Library.sentinel are accessible
2024-08-02 02:02:02.810582 WARNING: #0      _LibrarySentinel.noSuchMethod (package:dartdoc/src/model/library.dart:18:7)
2024-08-02 02:02:02.810582 WARNING: #1      _LibrarySentinel.element (package:dartdoc/src/model/library.dart:15:7)
2024-08-02 02:02:02.810582 WARNING: #2      SpecialClass.matchesClass (package:dartdoc/src/special_elements.dart:55:28)
2024-08-02 02:02:02.810582 WARNING: #3      SpecialClasses.addSpecial.<anonymous closure> (package:dartdoc/src/special_elements.dart:72:55)
2024-08-02 02:02:02.810582 WARNING: #4      IterableExtension.firstWhereOrNull (package:collection/src/iterable_extensions.dart:256:15)
2024-08-02 02:02:02.810582 WARNING: #5      SpecialClasses.addSpecial (package:dartdoc/src/special_elements.dart:72:29)
2024-08-02 02:02:02.810582 WARNING: #6      new Class (package:dartdoc/src/model/class.dart:46:33)
2024-08-02 02:02:02.810582 WARNING: #7      ModelElement._constructFromElementDeclaration (package:dartdoc/src/model/model_element.dart:281:25)
2024-08-02 02:02:02.810582 WARNING: #8      new ModelElement.for_ (package:dartdoc/src/model/model_element.dart:230:40)
2024-08-02 02:02:02.810582 WARNING: #9      new ModelElement.forElement (package:dartdoc/src/model/model_element.dart:106:25)
2024-08-02 02:02:02.810582 WARNING: #10     Nameable.getModelForElement (package:dartdoc/src/model/nameable.dart:66:20)
2024-08-02 02:02:02.810582 WARNING: #11     new ElementType.for_ (package:dartdoc/src/element_type.dart:44:37)
2024-08-02 02:02:02.810582 WARNING: #12     Nameable.getTypeFor (package:dartdoc/src/model/nameable.dart:93:19)
2024-08-02 02:02:02.810582 WARNING: #13     Mixin.superclassConstraints.<anonymous closure> (package:dartdoc/src/model/mixin.dart:21:13)
2024-08-02 02:02:02.810582 WARNING: #14     MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
2024-08-02 02:02:02.810582 WARNING: #15     ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
2024-08-02 02:02:02.810582 WARNING: #16     WhereIterator.moveNext (dart:_internal/iterable.dart:450:22)
2024-08-02 02:02:02.810582 WARNING: #17     new _GrowableList._ofOther (dart:core-patch/growable_array.dart:202:26)
2024-08-02 02:02:02.810582 WARNING: #18     new _GrowableList.of (dart:core-patch/growable_array.dart:152:26)
2024-08-02 02:02:02.810582 WARNING: #19     new List.of (dart:core-patch/array_patch.dart:39:18)
2024-08-02 02:02:02.810582 WARNING: #20     Mixin.superclassConstraints (package:dartdoc/src/model/mixin.dart:18:69)
2024-08-02 02:02:02.810582 WARNING: #21     Mixin.superclassConstraints (package:dartdoc/src/model/mixin.dart)
2024-08-02 02:02:02.810582 WARNING: #22     Mixin.inheritanceChain (package:dartdoc/src/model/mixin.dart:35:8)
2024-08-02 02:02:02.810582 WARNING: #23     Mixin.inheritanceChain (package:dartdoc/src/model/mixin.dart)
2024-08-02 02:02:02.810582 WARNING: #24     Class.inheritanceChain (package:dartdoc/src/model/class.dart:34:20)
2024-08-02 02:02:02.810582 WARNING: #25     Class.inheritanceChain (package:dartdoc/src/model/class.dart)
2024-08-02 02:02:02.810582 WARNING: #26     InheritingContainer._inheritedElements.<anonymous closure> (package:dartdoc/src/model/inheriting_container.dart:163:9)
2024-08-02 02:02:02.810582 WARNING: #27     InheritingContainer._inheritedElements (package:dartdoc/src/model/inheriting_container.dart:202:4)
2024-08-02 02:02:02.810582 WARNING: #28     InheritingContainer._inheritedElements (package:dartdoc/src/model/inheriting_container.dart)
2024-08-02 02:02:02.810582 WARNING: #29     InheritingContainer.inheritedMethods (package:dartdoc/src/model/inheriting_container.dart:110:35)
2024-08-02 02:02:02.810582 WARNING: #30     InheritingContainer.instanceMethods (package:dartdoc/src/model/inheriting_container.dart:343:37)
2024-08-02 02:02:02.810582 WARNING: #31     Container.allModelElements (package:dartdoc/src/model/container.dart:64:12)
2024-08-02 02:02:02.810582 WARNING: #32     InheritingContainer._allModelElements (package:dartdoc/src/model/inheriting_container.dart:104:14)
2024-08-02 02:02:02.810582 WARNING: #33     InheritingContainer._allModelElements (package:dartdoc/src/model/inheriting_container.dart)
2024-08-02 02:02:02.810582 WARNING: #34     InheritingContainer.allModelElements (package:dartdoc/src/model/inheriting_container.dart:287:46)
2024-08-02 02:02:02.810582 WARNING: #35     Class.allModelElements (package:dartdoc/src/model/class.dart:21:14)
2024-08-02 02:02:02.810582 WARNING: #36     Class.allModelElements (package:dartdoc/src/model/class.dart)
2024-08-02 02:02:02.810582 WARNING: #37     Library.allModelElements (package:dartdoc/src/model/library.dart:376:46)
2024-08-02 02:02:02.810582 WARNING: #38     Library.allModelElements (package:dartdoc/src/model/library.dart)
2024-08-02 02:02:02.810582 WARNING: #39     PackageGraph._gatherModelElements (package:dartdoc/src/model/package_graph.dart:917:36)
2024-08-02 02:02:02.810582 WARNING: #40     PackageGraph._precacheLocalDocs (package:dartdoc/src/model/package_graph.dart:179:28)
2024-08-02 02:02:02.810582 WARNING: #41     PackageGraph.initializePackageGraph (package:dartdoc/src/model/package_graph.dart:144:23)
2024-08-02 02:02:02.810582 WARNING: #42     PubPackageBuilder.buildPackageGraph (package:dartdoc/src/model/package_builder.dart:121:22)
2024-08-02 02:02:02.810582 WARNING: <asynchronous suspension>
2024-08-02 02:02:02.810582 WARNING: #43     Dartdoc.generateDocsBase (package:dartdoc/src/dartdoc.dart:189:24)
2024-08-02 02:02:02.810582 WARNING: <asynchronous suspension>
2024-08-02 02:02:02.810582 WARNING: #44     Dartdoc.generateDocs (package:dartdoc/src/dartdoc.dart:241:24)
2024-08-02 02:02:02.810582 WARNING: <asynchronous suspension>
2024-08-02 02:02:02.810582 WARNING: #45     Dartdoc.executeGuarded.<anonymous closure> (package:dartdoc/src/dartdoc.dart:273:9)
```

**Comment by srawlins** [comment]
Thanks for the report! I'm not sure about improving the error, but this issue is fixed with a fix already in the works for another issue. https://github.com/dart-lang/dartdoc/issues/3826 🎉
